third-culture-software/bhima
# Budget title subtotals

Compute the budget, actuals, etc for all title accounts.  Note that these are computed on the fly and are based entirely on child expense and income accounts, not child title accounts. 

Closes https://github.com/IMA-WorldHealth/bhima/issues/7441

*TESTING*
- Use bhima_test in your dev environment (do `npm run build; npm run build:db' first)
- If you do not see `Budget Management` under `Finances`, update your role permissions.
- In `Finances > BudgetManagement [MENU]' execute `Expor tYTD Budget Data (CSV)`
- Using Excel or a text editor, edit the CSV file and add budget amounts for various income and expense accounts.
   - Be sure to add some budget data to account `70111010 - Vente Médicaments en comprimes`
   - Any data entered in title accounts will be ignored
- Run BHIMA again
   - When you go to the `Finance > Budget Management` page, you will see the accounts but no budget or actuals data
   - To add Actuals expenses to help verify the totals do this
      - Go to `Finance > Simple Vouchers`
      - For Transaction Type, select "Purchases"
      - Add a description
      - For Debit, select account 60310014
      - For Credit, select account 31110011
      - Set an amount (such as 100)
      - Click on [Submit]
   - go to `Finance > Journal`
      - Change the Search to "All Time"
      - Select all transactions
      - Click on the 'Trial Balance' button (near the right end of button row at the top)
      - A confirmation dialog will appear, click on [Submit]
   - Return to the `Finance > Budget Management` page.  Now you will see account 7011010 has some actuals data
   - In the the `Finance > Budget Management [MENU]`, execute `Import Budget CSV file` and select the CSV file you modified earlier.   When it is imported, you should see data for title accounts (where the data is no zero). 
   - Click on the top-right collapse button to show only the top-level accounts.   This should make it simple to see if overall computations are correct.  Here is what this looks like for me (probably with different budget numbers that you will use; also I fixed missing items in the summary row after this screenshot): 
![image](https://github.com/IMA-WorldHealth/bhima/assets/62145/0feca702-fba7-47e9-aac4-5a874e83ceba)
   - With all rows expended, verify the title computations.  Here is what the first part of the income section looks like for me: 
![image](https://github.com/IMA-WorldHealth/bhima/assets/62145/119b05a4-c275-4326-9d98-ff2e4ae0e9e2)
   - Once you have budget and actuals data, verify that the CSV output and the PDF report are correct.

- It would probably be a good idea to test this with a production database.
